resources/ext/cli/config-migration

```diff
@@ -0,0 +1,243 @@
+}
+
+# This function copies the necessary PEM files from Puppet to the PuppetDB
+# SSL directory.
+#
+# This expects various environment variables to have already been set to work.
+#
+# Note: This does not respect directory structures, and will move files blindly
+function migrate_files() {
+  destination=$1
+  orig_file=$2
+  if [ ! -e "$orig_file" ]; then
+    echo "Warning: Unable to find all ini configuration files to copy"
+    echo
+    echo "  This tool requires the following files to exist:"
+    echo
+    echo "  * $orig_file"
+    echo
+    echo "  These files may be missing due to the fact that your host's Puppet"
+    echo "  certificates may not have been signed yet, probably due to the"
+    echo "  lack of a complete Puppet agent run. Try running puppet first, for"
+    echo "  example:"
+    echo
+    echo "      puppet agent --test"
+    echo
+    echo "  Afterwards re-run this tool then restart PuppetDB to complete the SSL"
+    echo "  setup:"
+    echo
+    echo "      ${migration_command} -f"
+    exit 1
+  fi
+  if [ -e "${destination}/${orig_file##*/}" ] ; then
+    if $force ; then
+      echo "Replacing file: ${destination}/${orig_file##*/} with ${orig_file}"
+      cp -p "$orig_file" "${destination}/${orig_file##*/}"
+    else
+      echo "${destination}/${orig_file##*/} already exists. To replace with original file, rerun ${migration_command} -f"
+    fi
+  else
+    echo "Copying file: ${orig_file} to ${destination}"
+    mkdir -p "$destination"
+    cp -p "$orig_file" "${destination}/${orig_file##*/}"
+  fi
+}
+
+# This function copies the necessary KahaDB queue from the old varlib dir to the
+# new directory.
+#
+# $1 - original location to be migrated
+# $2 - migration target location
+#
+# This expects various environment variables to have already been set to work.
+function migrate_directory() {
+  old_location="$1"
+  new_location="$2"
+  if [ ! -e "$old_location" ]; then
+    echo "Warning: Unable to find directory to copy"
+    echo
+    echo "  This tool requires the following directory to exist:"
+    echo
+    echo "  * $old_location"
+    echo
+    echo "  These files may be missing due to the fact that your host's Puppet"
+    echo "  certificates may not have been signed yet, probably due to the"
+    echo "  lack of a complete Puppet agent run. Try running puppet first, for"
+    echo "  example:"
+    echo
+    echo "      puppet agent --test"
+    echo
+    echo "  Afterwards re-run this tool then restart PuppetDB to complete the SSL"
+    echo "  setup:"
+    echo
+    echo "      ${migration_command} -f"
+    exit 1
+  fi
+  mkdir -p "${new_location}"
+  echo "Copying files from ${old_location} to ${new_location}"
+  cp -pr $old_location/* $new_location/
+}
+
+
+########
+# MAIN #
+########
+
+# Gather command line options
+while getopts "fh" opt;
+do
+  case $opt in
+    f)
+      force=true ;;
+    h)
+      usage ;;
+    *)
+      usage ;;
+  esac
+done
+
+${force:=false}
+
+set -e
```

Comment:
You might have a good reason not to, but if not, I'd suggest putting this at the top.